src/units/zone_server.rs

```diff
@@ -145,7 +145,17 @@ impl ZoneServerUnit {
         let svc = MandatoryMiddlewareSvc::<_, _, ()>::new(svc);
         let svc = Arc::new(svc);
 
-        for addr in self.listen.iter().cloned() {
+        let listen_strings = self
+            .listen
+            .iter()
+            .filter_map(|addr| match addr {
+                ListenAddr::Udp(socket_addr) => Some(socket_addr.to_string()),
+                ListenAddr::Tcp(socket_addr) => Some(socket_addr.to_string()),
+                ListenAddr::UdpSocket(_) | ListenAddr::TcpListener(_) => None,
```

Comment:
If I recall correctly, this is for printing where it listens so a TODO here to get the addr and print it might be nice. I can imagine that it's confusing if the application is telling you that it is listening on a subset of addrs than it actually is.